### DISCUSSION TITLE
DNS Leak!?

### DISCUSSION THREAD

**Discussion by collingerac** [qdm12/gluetun#2119] [Q&A]
What should I do to fix this if anything? 
WARN [dns] ⚠️⚠️⚠️  keeping the default container nameservers, this will likely leak DNS traffic outside the VPN and go through your container network DNS outside the VPN tunnel!
Thanks for the help.

**Comment by ezekieldas** [comment]
are you using `DNS_KEEP_NAMESERVER` ? See:  https://github.com/qdm12/gluetun-wiki/blob/main/setup/options/dns.md: 

This means dns is referenced as the container's host defined resolvers (eg, linux /etc/resolv.conf). This is Docker behavior, not Gluetun. 

If you're running a resolver which attaches to Gluetun then you're probably fine. You can use this to determine leakage: https://dnscheck.tools
